docs2/site/docs/migrations/migration4.md

```diff
@@ -2,105 +2,291 @@
+
 ## Breaking Changes
 
-* GraphQL.NET now uses GraphQL-Parser v7 with new memory model
-* `ExecutionResult.Data` format breaking changes. Now objects are serialized as `ObjectProperty` arrays, not object dictionaries.
-  Both `GraphQL.NewtonsoftJson` and `GraphQL.SystemTextJson` serializers received the necessary changes to produce the same JSON as before.
-  However, consumers using `ExecutionResult` instances directly most likely will not work correctly.
-* `NameConverter`, `SchemaFilter` and `FieldMiddleware` have been removed from `ExecutionOptions` and are now properties on the `Schema`.
-  These properties can be set in the constructor of the `Schema` instance, or within your DI composition root, or at any time before
-  any query is executed. Once a query has been executed, changes to these fields is not allowed, and adding middleware via the field middleware
-  builder has no effect.
-* The signature of `IFieldMiddlewareBuilder.Use` has been changed to remove the schema from delegate. Since the schema is now known, there is no
+### Schema Configuration Options Moved
+
+`NameConverter`, `SchemaFilter` and `FieldMiddleware` have been removed from `ExecutionOptions` and are now properties on `Schema`.
+These properties can be set in the constructor of the `Schema` instance, or within your DI composition root, or at any time before
+any query is executed. Once a query has been executed, changes to these fields is not allowed, and adding middleware via the field middleware
+builder has no effect.
+
+### Middleware Builders
+
+- The signature of `IFieldMiddlewareBuilder.Use` has been changed to remove the schema from delegate. Since the schema is now known, there is no
   need for it to be passed to the middleware builder.
-* The middleware `Use<T>` extension method has been removed. Please use the `Use` method with a middleware instance instead.
-* `GraphQL.Utilities.ServiceProviderExtensions` has been made internal. This affects usages of its extension method `GetRequiredService`. Instead, reference the `Microsoft.Extensions.DependencyInjection.Abstractions` NuGet package and use extension method from `Microsoft.Extensions.DependencyInjection.ServiceProviderServiceExtensions` class.
-* `GraphQL.Instrumentation.StatsReport` and its associated classes have been removed. Please copy the source code into your project if you require these classes.
-* When used, Apollo tracing will now convert the starting timestamp to UTC so that `StartTime` and `EndTime` are properly serialized as UTC values.
-* `ApolloTracing.ConvertTime` is now private and `ResolverTrace.Path` does not initialize an empty list when created.
+- The middleware `Use<T>` extension method has been removed. Please use the `Use` method with a middleware instance instead.
+
+See [Field Middleware](https://graphql-dotnet.github.io/docs/getting-started/field-middleware) for more information.
```

Comment:
This page needs fixing.

Reply:
What's wrong with it?  It was updated inside #2210 along with the relevant changes.

Reply:
I forgot to switch to `develop`. Everything is good but link to directives in https://github.com/graphql-dotnet/graphql-dotnet/blob/develop/docs2/site/docs/getting-started/field-middleware.md#field-middleware-vs-directive section is broken.

Reply:
Good catch.  Should probably update that with the rest of your changes to directive documentation.